fix: 🚑 room.entity파일의 위도,경도 오류수정

seokimun opened the pull request:
- 위도,경도에서 소수점이 표현안되는 오류수정

seokimun committed:
fix: 🚑 room.entity파일의 위도,경도 오류수정
- 위도,경도에서 소수점이 표현안되는 오류수정

seokimun committed:
Merge remote changes

seokimun committed:
깃 오류 해결

seokimun committed:
Resolve merge conflicts

seokimun committed:
Resolve merge

seokimun committed:
feat: ✨ 원룸리스트 조회 API 개발
- 유저의 대학과 일치한 대학원룸 리스트를 불러오고, 즐겨찾기 유무도 확인하여 원룸리스트 출력

seokimun committed:
feat: ✨ 특정대학의 필터링된 원룸조회 API 개발
- 특정대학의 월세와 전세의 범위를 설정하여 해당 범위내의 대학원룸목록을 출력

woomae commented on src/api/v1/room/room.service.ts:
```diff
@@ -1,4 +1,62 @@
+                (fav) => fav.room._id === room._id,
+            );
+
+            const plainRoom = instanceToPlain(room);
```
객체화 하면 안됨

woomae commented on src/api/v1/room/room.service.ts:
```diff
@@ -1,4 +1,62 @@
+            providerId,
+        );
+
+        return rooms.map((room) => {
```
리턴문이 너무 길어져서 함수로 빼는것도 좋을 듯 합니다

woomae commented on src/api/v1/room/room.service.ts:
```diff
@@ -1,4 +1,62 @@
+        depositRange?: [number, number],
+        costRange?: [number, number],
+    ): Promise<RoomResponseDto[]> {
+        let rooms = await this.roomRepository.findByUniversityName(
```
let 왜나온건가용?

woomae commented on src/api/v1/room/room.repository.ts:
```diff
@@ -0,0 +1,28 @@
+    constructor(
+        @InjectRepository(Room) private readonly repository: Repository<Room>,
+    ) {
+        super(repository.target, repository.manager, repository.queryRunner);
```
빼도 상관없지않나요?

seokimun committed:
refator: 🎨 원룸조회 API 코드개선
- room컨트롤러에서 코드의 가독성과 유지보수성 향상
- 단일책임의 원칙으로 roomFilter파일 추가
- 폴더구조화
- 서비스 폴더의 코드 가독성과 유지보수성 향상

seokimun committed:
refactor: ♻️ 원룸조회 API 코드 리팩토링
- 레포지토리 폴더삭제
- room서비스 폴더삭제

seokimun committed:
refactor: ♻️ room서비스, 레포지토리 코드 리팩토링
- filter서비스 삭제
- room서비스 코드 최적화

seokimun committed:
refactor: ♻️ 원룸조회 API 코드개선
- DTO 파일삭제
- 확실한 책임분할 코드개선

seokimun committed:
refactor: ♻️ 원룸조회 API 코드개선
- 서비스 코드가독성 향상
- 레포지토리 쿼리문 개선
- dto파일 추가

seokimun committed:
fix: 🐛 imageurl 버그수정
- imageurl이 하나만 출력되도록 버그수정

seokimun committed:
refactor: ♻️ 레포지토리 파일수정
- if문 삭제
- 각 클래스화 해줌

SeongHo5 commented:
코드 리뷰 문화 👍

SeongHo5 commented on src/api/v1/mock/mock.controller.ts:
```diff
@@ -12,14 +12,14 @@ import { MockService } from './mock.service';
+    //특정대학 원룸조회
     @Get('rooms')
     async findRoomsByUniversitylName(
         @Query('university_name') university_name: string,
```
Query Parameter나 Request Body는 camelCase를 사용하는 게 좋아요
보면 좋은 레퍼런스: https://jojoldu.tistory.com/783

SeongHo5 commented on src/api/v1/mock/mock.controller.ts:
```diff
@@ -12,14 +12,14 @@ import { MockService } from './mock.service';
+    //특정원룸 조회
     @Get('rooms/:id')
     async findOneByRoomId(@Param('id', ParseIntPipe) id: number) {
         const room = await this.mockService.findOneByRoomId(id);
         return room;
```
변수를 선언하고 처리하는 로직이 없다면, 불필요한 선언 없이 바로 반환하는 건 어떨까요
```suggestion
        return await this.mockService.findOneByRoomId(id);

SeongHo5 reviewed (commented):
김기문 화이팅

(근데 타입스크립트인데 타입 정의한 부분 찾기가 힘드네요 ㅎㅎ;;)
